modules/broker-amqp/src/main/java/io/ballerina/messaging/broker/amqp/rest/ConnectionsApiDelegate.java

```diff
@@ -76,18 +77,29 @@ public Response getAllConnections(Subject subject) {
      * Forces disconnection of the AMQP connection.
      *
      * @param id      connection identifier
+     * @param force   indicates if the connection should be closed forcefully or not
      * @param subject authentication subject containing user information of the user that has invoked the API
      * @return HTTP/1.1 202 accepted with {@link RequestAcceptedResponse}
      */
-    public Response closeConnection(int id, Subject subject) {
+    public Response closeConnection(int id, boolean force, Subject subject) {
         try {
             authHandler.handle(ResourceAuthScope.CONNECTIONS_CLOSE, subject);
-            connectionManager.forceDisconnect(id, "Connection close request received from REST API.");
-            return Response.accepted()
-                           .entity(new RequestAcceptedResponse().message(
-                                   "Forceful disconnection of connection " + id + " accepted."))
-                           .build();
-        } catch (NoSuchElementException e) {
+            if (!force) {
```

Comment:
Shall we swap the conditional blocks and have the condition as `force`. That's more readable I guess. WDYT?